src/com/linkedin/parseq/TaskDescriptor.java

```diff
@@ -0,0 +1,23 @@
+package com.linkedin.parseq;
+
+/**
+ * An API to provide description for task.
+ *
+ * parseq-lambda-names provide an implementation for this interface using Java SPI. Any changes made to this interface
+ * might require a change to implementation in parseq-lambda-names.
+ *
+ * The implementation doesnt need to be thread-safe.
+ *
+ * @author Siddharth Sodhani (ssodhani@linkedin.com)
+ */
+public interface TaskDescriptor {
```

Comment:
As discussed, copying TaskDescriptor to contrib parseq-lambda-names project and removing dependency on parseq. Testing in voyager-api.